File: java_db_service/src/main/java/ru/mipt/osint_db_service/model/webParsing/WebSourcesLink.java
package ru.mipt.osint_db_service.model.webParsing;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "web_sources_links", schema = "osint_web")
public class WebSourcesLink {
    @Id
    @Column(name = "id", nullable = false)
    private String id;

    @Column(name = "link", nullable = false, length = Integer.MAX_VALUE)
    private String link;

    @Column(name = "analyzed_time", length = Integer.MAX_VALUE)
    private String analyzedTime;

}

File: Osint_web_service/src/main/kotlin/ru/bitok/osint/osint_web_service/entity/WebSourcesLink.kt
package ru.bitok.osint.osint_web_service.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.ColumnDefault
import java.time.Instant

@Entity
@Table(name = "web_sources_links", schema = "osint_web_v2")
class WebSourcesLink {
    @Id
    @Column(name = "link", nullable = false, length = Integer.MAX_VALUE)
    var link: String = ""

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "analyzed_time")
    var analyzedTime: Instant = Instant.now()

    @Column(name = "active", nullable = false)
    var active: Boolean = true

    @ColumnDefault("NULL::character varying")
    @Column(name = "user_comment")
    var userComment: String? = null
}